Moved documentation to Hugo

brettansley opened the pull request:
Moved to using Hugo to generate the documentation - this makes it easier to
manage because content is stored as markdown and rendered by Hugo
rather than the previous version of the docs which were all hand-crafted HTML. 
Hugo can be found here: https://gohugo.io/

Commit:
Moved documentation to Hugo

Used Hugo to generate the documentation - this makes it easier to
manage because content is stored as markdown and rendered by Hugo
rather than the previous version which was all hand-crafted HTML. Hugo
can be found here: https://gohugo.io/

Commit:
Removed unnecessary wercker file

The original project used wercker for deployment. This file is
unnecessary and has been removed.

Commit:
Config changes for CF deployment

2 configuration changes were needed for successful deployment.
Changed the manifest file to include the correct domain and application
names.
Edited the Staticfile to remove unnecessary content.

mtekel commented:
Few comments:
1. To review, edit manifest, configure correct domain of your env., then push.
2. Did you write this all @brettansley  or is this borrowed from somewhere? I have noticed multiple language buildpack in the documentation, but when we tried that one with @Jonty we had on luck. Perhaps we should not encourage people to use it...

Commit:
Updated the cf-ssh page

The cf-ssh page had an 18F specific instruction - this has been removed
to keep it more generic.

Commit:
Added the list of tasks for people to try

Added the list of tasks that we want people to try out during the trial.

Commit:
Regenerated all files using Hugo

Regenerated all the site files based on the changes made to the mark
down files in previous commits.

Commit:
Regenerate last files.

Used Hugo to regenerate again to include the final changes.

mtekel commented on CloudFoundry/content/apps/flask.md:
````diff
@@ -0,0 +1,125 @@
+Switching your Flask app to serve with Waitress instead of the built-in server is fairly simple. The first thing you'll want to do is the app's port to the `VCAP_APP_PORT` environment variable. This is set by Cloud Foundry for you so you don't have to worry about it, but you _do_ need to let Flask know about it. And while you're developing locally you might want to set a local `VCAP_APP_PORT`. Yo ...
+
+```python
+port = port = int(os.getenv("VCAP_APP_PORT"))
````
Looks like this should be just `port = int(os.getenv("VCAP_APP_PORT"))`

brettansley replied:
You're right - I'll update that

mtekel commented on CloudFoundry/content/apps/static.md:
```diff
@@ -0,0 +1,66 @@
+applications:
+- name: my-static-site
+  memory: 64M
+  buildpack: https://github.com/cloudfoundry/staticfile-buildpack.git
```
We already provide `staticfile_buildpack` with platform. Perhaps we want people to use that one instead of external buildpack?

brettansley replied:
I'll re-write that a little to instruct people to create an empty Staticfile to use the provided buildpack

mtekel commented on CloudFoundry/content/apps/sample-apps.md:
```diff
@@ -0,0 +1,95 @@
+You should now find that both instances of the app return the same data every time. The load balancer will round-robin connections between the different instances, so hitting shift-refresh in your browser a couple of times will switch between deployed instances.
+
+
+ ### Flask SQLAlchemy Sample App:
```
This doesn't highlight properly. Perhaps remove the leading space?

mtekel commented on CloudFoundry/content/apps/sample-apps.md:
````diff
@@ -0,0 +1,95 @@
+ A simple microblog app built using Python/Flask using SQLAlchemy to support a postgres DB backend. The default **username / password** for the flask-sqlalchemy app is **admin / default**
+
+ ``` bash
+ git clone https://github.com/alphagov/flask-sqlalchemy-postgres-heroku-example.git
````
Leading space here looks like typo. It doesn't look nice in the code highlighting block on the rendered web site.

mtekel commented on CloudFoundry/content/apps/quotas.md:
```diff
@@ -0,0 +1,114 @@
+- 1000 application routes.
+- 10.024GB of RAM.
+- 100 service instances.
+
```
Why repeat quotas in text and again in the list again? I'd just leave the list and fix `10.024`GB of ram to `10`GB of RAM

mtekel commented on CloudFoundry/content/apps/quotas.md:
```diff
@@ -0,0 +1,114 @@
+- 100 service instances.
+
+###### Routes:
+
```
The five `#` make all the titles (routes, ram etc.) look much smaller than the text of the paragraphs. I'd just use 4 hashes like with Quotas (line 9).

mtekel commented on CloudFoundry/content/apps/quotas.md:
```diff
@@ -0,0 +1,114 @@
+	- Guaranteed at least 10% vCPU time. 
+	- Offered up to 100% vCPU time. 
+	- Limited to 375MB of RAM.
+
```
I'm not 100% sure if this is how CPU allocation works. I think it depends on the ratio of the container memory to the actual DEA memory, which tenants might not know. The number of CPUs on DEA also impacts how much actual compute might a container get. This is also not known to the tenants....

Commit:
Removed the Continuous Deployment page

Removed 18F’s CD page as we haven’t tested some of what is mentioned on
our platform.

Commit:
Corrected a typo on the flask page

Removed a redundant ‘port =‘ from the flask page.

Commit:
Removed reference to multi-buildpacr deploys

We have tested multi-buildpack deploys and have not been able to get
them to work so we have removed this section.

Commit:
Corrected formatting and wording

Corrected to formatting and working of the quotas markdown.

Commit:
Removed leading spaces

There were a couple of leading spaces that were causing formatting
issues on this page. These have been corrected.

Commit:
Changed to use the default static site buildpack

Updated the instructions to have people using the default static site
buildpack that is included in the CF platform.

Commit:
Tidied up the layout of the index page and support link

Made the support sentence more readable on the index page then
repointed the support link on the menu bar back to the index page
support section and regenerated all pages to match.

Commit:
Corrected the link to the list of things to try

At the bottom of the sample apps page there is a link to the list of
things we want people to try - this link was pointing to the wrong
location.

Commit:
Buildpack changes

Added a reference to the ‘cf buildpacks’ command in the concepts page
and added the ‘buildpack: static_buildpack’ configuration line to the
sample YAML.

mtekel commented:
Looks fine, I have no more comments...
